common/client/node.go

```diff
@@ -43,6 +44,14 @@ type NodeConfig interface {
 	SelectionMode() string
 	SyncThreshold() uint32
 	NodeIsSyncingEnabled() bool
+	FinalizedBlockPollInterval() time.Duration
```

Comment:
Just wondering, do we need different configs for each type of poll?
What if we just reuse the PollInterval for all polling?

Reply:
In some chains, we may also look for new heads via polling, not via subscribe.
In that case too, we wouldn't like to poll separately for new heads and new finalized heads. Mostly just make a single batch call to get both.
So that's why I am thinking, could we club all things to be polled under a same config, and fetch them together in a batch call?

Reply:
I have a similar impression, it would be more efficient to batch whenever we can instead of introducing a new ticker that will increase pressure on the RPC. AFAIK we already poll RPC to verify if it's healthy so probably we could use that logic for adding latest finalized block. If we manage to bundle that together into a single batch then we will get finality tracking for free

Maybe reusing existing `<-pollCh`?

```go
	for {
		select {
		case <-n.nodeCtx.Done():
			return
		case <-pollCh:

```

Reply:
I'm not in favour of merging all of the polls into a single ticker, as they check different properties of the RPC.
Poll checks that RPC is reachable, this is super basic check and we want to do it often and be aggressive with the timeouts, if an RPC needs > 1s to return it's version, it not healthy, while for finalized block it seems ok. 

Regarding the new heads polling, it makes sense to batch poll in this case